src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemDatasetValidator.cs

```diff
@@ -0,0 +1,123 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Store;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    /// <summary>
+    /// Provides functionality to validate a <see cref="DicomDataset"/> to make sure it meets the minimum requirement when Adding.
+    /// </summary>
+    public class AddWorkitemDatasetValidator : IAddWorkitemDatasetValidator
+    {
+        public void Validate(DicomDataset dicomDataset, string workitemInstanceUid)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+
+            ValidateRequiredTags(dicomDataset, workitemInstanceUid);
+        }
+
+        private static void ValidateRequiredTags(DicomDataset dicomDataset, string workitemInstanceUid)
+        {
+            // The format of the identifiers will be validated by fo-dicom.
+            string workitemUid = EnsureRequiredTagIsPresent(dicomDataset, DicomTag.AffectedSOPInstanceUID);
+
+            // If the workitemInstanceUid is specified, then the workitemUid must match.
+            if (!string.IsNullOrWhiteSpace(workitemInstanceUid) &&
```

Comment:
We also need to validate that the state is SCHEDULED and not other things

Reply:
Done